From 50a0554d12cff58b3ffbd51d3194304244b87023 Mon Sep 17 00:00:00 2001
From: Krunal Soni <ksoni@codeaurora.org>
Date: Tue, 3 Oct 2017 19:39:48 -0700
Subject: qcacld-3.0: Do boundary check on num_vdev_mac_entries param

Current doesn't perform boundary check on num_vdev_mac_entries param
which coming from firmware. Without boundary check, driver may be
exposed to buffer overflow.

Check against the boundary limit before using it.

CRs-Fixed: 2119430
Change-Id: I502926a7f783acc7b73a3fbbbd70386a099b48b3
---
 core/wma/src/wma_main.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/wma/src/wma_main.c b/core/wma/src/wma_main.c
index 63d9f80..e1b5bb5 100644
--- a/core/wma/src/wma_main.c
+++ b/core/wma/src/wma_main.c
@@ -3093,6 +3093,11 @@ static int wma_pdev_set_hw_mode_resp_evt_handler(void *handle,
 		 */
 		goto fail;
 	}
+	if (param_buf->fixed_param->num_vdev_mac_entries >=
+						MAX_VDEV_SUPPORTED) {
+		WMA_LOGE("num_vdev_mac_entries crossed max value");
+		goto fail;
+	}
 
 	wmi_event = param_buf->fixed_param;
 	hw_mode_resp->status = wmi_event->status;
-- 
cgit v1.1

